c2c-connector: refactoring

Danielius1922 opened the pull request.

Danielius1922 committed:
c2c-connector: refactoring

  - update configuration to be loadable from yaml

Danielius1922 committed:
Minor configuration updates

jkralik commented on pkg/security/oauth/manager/manager.go:
```diff
@@ -2,17 +2,15 @@ package manager
 
```
OAuth manager is still needed?

Danielius1922 replied:
I think it can be removed, but I want to fix re-enable and fix all tests first before I start removing code.

Danielius1922 committed:
test: refactor helper methods

  - create GetDefaultDeviceAuthorizationCode and GetDefaultServiceToken
functions
  - add host parameter to NewRequest function

Danielius1922 committed:
Move coap-gw authorization to pkg/security/oauth2

Danielius1922 committed:
c2c-connector: refactor getDevices related code

Danielius1922 committed:
c2c-connector: refactor retrieve and update resource

Danielius1922 committed:
c2c-gw: fix superfluos WriteHeader call error

Danielius1922 commented on cloud2cloud-connector/service/requestHandler.go:
```diff
@@ -52,25 +43,19 @@ func logAndWriteErrorResponse(err error, statusCode int, w http.ResponseWriter)
 		provisionCache: cache.New(5*time.Minute, 10*time.Minute),
 		triggerTask:    triggerTask,
-		ownerClaim:     ownerClaim,
+		ownerClaim:     "sub",
```
TODO: remove ownerClaim

Danielius1922 committed:
Remove obsolete environmental variables

Danielius1922 committed:
c2c-connector: parse owner from request context

Danielius1922 committed:
c2c-connector: refresh token for origin and target clouds

Danielius1922 committed:
c2c-connector: fix linkedAccounts_test.go

jkralik commented on cloud2cloud-connector/config.yaml:
```diff
@@ -0,0 +1,87 @@
+    authorization:
+      clientID: ""
+      clientSecret: ""
+      deviceIdClaim: ""
```
`deviceIdClaim` is not needed for c2c-connector

jkralik commented on cloud2cloud-connector/config.yaml:
```diff
@@ -0,0 +1,87 @@
+      clientCertificateRequired: false
+    eventsURL: ""
+    pullDevices:
+      disabled: false
```
pls change it to `enabled: true`

jkralik commented on cloud2cloud-connector/config.yaml:
```diff
@@ -0,0 +1,87 @@
+        keyFile: "/secrets/private/cert.key"
+        certFile: "/secrets/public/cert.crt"
+        useSystemCAPool: false
+  resourceDirectory:
```
change it to `grpcGateway`

Danielius1922 committed:
c2c-connector: use gRPC client instead of RD client

Danielius1922 committed:
c2c-connector: lower cyclomatic complexity

jkralik reviewed (approved):
Good job!
